src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java

```diff
@@ -103,6 +108,22 @@ public static boolean isStringSafe(@CheckForNull String input, @CheckForNull Env
                 : true;
     }
 
+    /** Restrict stored arguments to a reasonable subset of types so we don't retain totally arbitrary objects
+     *  in memory. Generally we aim to allow storing anything that maps correctly to an {@link UninstantiatedDescribable}
+     *  But we may allow a few extras doesn't address, if they're safe to store.
+     */
+    boolean isStoreableType(Object ob) {
+        if (ob == null) {
+            return true;
+        } else if (ob instanceof CharSequence || ob instanceof Number || ob instanceof Boolean
+                || ob instanceof Map || ob instanceof List || ob instanceof UninstantiatedDescribable
+                || ob instanceof URL || ob instanceof Result || ob instanceof Exception) {
+            return true;
+        }
+        Class c = ob.getClass();
+        return c.isPrimitive() || c.isEnum() || c.isArray();
```

Comment:
So you are allowing `EvilObject[]` ?

Reply:
@rsandell IIRC, it gets coerced to a List internally (see https://github.com/jenkinsci/structs-plugin/blob/8d4446fe64ce79fa476345d49cdbb100307af76c/plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableModel.java#L262) but... y'know, the way DescribableModels work is kind of black magic, so better safe than sorry and I'll add Array sanitization.

Reply:
aand, done @rsandell